force-app/main/default/lwc/similarProducts/__tests__/similarProducts.test.js

```diff
@@ -0,0 +1,106 @@
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('displays a list of product tiles when the Apex wire adapter returns data', () => {
+        const element = createElement('c-similar-products', {
+            is: SimilarProducts
+        });
+        document.body.appendChild(element);
+
+        // Emit data from getRecord adapter
+        getRecordAdapter.emit(mockGetRecord);
+
+        // Emit Data from the Apex wire adapter.
+        getSimilarProductsListAdapter.emit(mockSimilarProducts);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const productListItemEl = element.shadowRoot.querySelector(
+                'c-product-list-item'
+            );
+            expect(productListItemEl).not.toBeNull();
+        });
+    });
+
+    it('displays a placeholder when no similar products are returned', () => {
+        const element = createElement('c-similar-products', {
+            is: SimilarProducts
+        });
+        document.body.appendChild(element);
+
+        // Emit data from getRecord adapter
+        getRecordAdapter.emit(mockGetRecord);
+
+        // Emit Data from the Apex wire adapter.
+        getSimilarProductsListAdapter.emit([]);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const placeholderEl = element.shadowRoot.querySelector(
+                'c-placeholder'
+            );
+            expect(placeholderEl).not.toBeNull();
+        });
+    });
+
+    it('displays a n error panel when the Apex wire adaptor returns an error', () => {
+        const element = createElement('c-similar-products', {
+            is: SimilarProducts
+        });
+        document.body.appendChild(element);
+
+        // Emit data from getRecord adapter
+        getRecordAdapter.emit(mockGetRecord);
+
+        // Emit an error from the Apex wire adapter.
+        getSimilarProductsListAdapter.error();
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const errorPanelEl = element.shadowRoot.querySelector(
+                'c-error-panel'
+            );
+            expect(errorPanelEl).not.toBeNull();
```

Comment:
Similar to other component, would be nice to do a `falsy` check that `errors` is passed as value to the error panel `errors` property.